src/app/control.rs

```diff
@@ -263,10 +268,10 @@ pub mod resolve {
             let target = client::Target {
                 addr,
                 server_name: dst.identity.clone(),
-                log_ctx: logging::admin().client("control", dst.addr.clone()),
             };
 
-            State::Inner(mk_svc.call(target))
+            info_span!("control", peer_addr=%addr, peer_identity=?dst.identity.clone())
+                .in_scope(move || State::Inner(mk_svc.call(target).instrument(Span::current())))
```

Comment:
Nit/TIOLI: if we update to `tracing-futures` 0.1 (which we should do anyway) there is [a convenience method](https://docs.rs/tracing-futures/0.1.0/tracing_futures/trait.Instrument.html#method.in_current_span) for this:
```suggestion
                .in_scope(move || State::Inner(mk_svc.call(target).in_current_span()))
```